Allow same ID to be used

yuliswe opened the pull request:
### Description
Adds a removeClient call when creating a new client with an existing ID

**Asana link**:
https://app.asana.com/0/1200399401331050/1209320387692695

### Test plan (what/how did I test?)
Manual

Commit:
Allow same ID to be used

Commit:
Add logs

Commit:
Fix GetClientIdFromHostname for localhost

Commit:
Update regex

amahalwy commented on server.js:
```diff
@@ -18,6 +20,10 @@ export default function (opt) {
   const landingPage = opt.landing || 'https://localtunnel.github.io/www/';
 
   function GetClientIdFromHostname(hostname) {
+    // The myTldjs.getSubdomain() function will return null for localhost
+    if (hostname.match(/\.localhost(:\d+)?$/g)) {
+      return hostname.split('.')[0];
+    }
```
Is this only for your local testing? Do we need this committed

yuliswe replied:
It is for running it using localhost.
It's nice to have IMO. I don't believe it will break anything.

amahalwy commented on server.js:
```diff
@@ -53,6 +59,36 @@ export default function (opt) {
+    const clientId = ctx.params.id;
+    const client = manager.getClient(clientId);
+    if (!client) {
+      ctx.throw(404);
```
```suggestion
      ctx.throw(405);
```

amahalwy commented on server.js:
```diff
@@ -53,6 +59,36 @@ export default function (opt) {
+      manager.removeClient(clientId);
+    } catch (e) {
+      logger.error(e);
+      ctx.throw(404);
```
```suggestion
      ctx.throw(405);
```

amahalwy commented on server.js:
```diff
@@ -53,6 +59,36 @@ export default function (opt) {
+    const method = ctx.request.method;
+    const path = ctx.request.path;
+    const query = ctx.request.query;
+    logger.debug('Incoming request', method, path, JSON.stringify(query));
```
Do we need this for the eventual live server code?

amahalwy commented on server.js:
```diff
@@ -53,6 +59,36 @@ export default function (opt) {
   app.use(router.routes());
   app.use(router.allowedMethods());
 
+  router.del('/api/tunnels/:id', async (ctx, next) => {
```
Will we keep this just in case or good to drop?

yuliswe replied:
It's nice to have.

Commit:
Use endOrDestroy on socket

Socket connections need to be properly closed by calling .end().

Using .destroy() on a socket would cause the client side not receive the FIN pack and not exit.

Commit:
Change 404 -> 405

Commit:
fix
